src/load.c

```diff
@@ -451,10 +451,7 @@ static int load_module(xmp_context opaque, HIO_HANDLE *h)
 	test_result = load_result = -1;
 	for (i = 0; format_loaders[i] != NULL; i++) {
 		hio_seek(h, 0, SEEK_SET);
-
-		if (hio_error(h)) {
-			/* reset error flag */
-		}
+		hio_error(h); /* reset error flag */
```

Comment:
This appears to have been intentional (don't know what purpose it served though). I think this change is the correct thing to do but I just thought that was interesting.
https://github.com/libxmp/libxmp/commit/42d4717802a9e10c41b117cd692de902e58ef6d5